### ISSUE TITLE
Add IDBFactory.databases() method to lib.DOM

### ISSUE THREAD

**Issue by nathanblair** [microsoft/TypeScript#36706]
## Search Terms

idbfactory databases

## Suggestion

https://github.com/microsoft/TypeScript/blob/fa3173f8f6443307739fc7c41f2bdd680f56a844/lib/lib.dom.d.ts#L9386

Should be adjusted to include the databases() method specified in:
- https://developer.mozilla.org/en-US/docs/Web/API/IDBFactory/databases
- https://w3c.github.io/IndexedDB/#dom-idbfactory-databases

## Use Cases

Getting a list of all IndexedDB databases controlled by a window

## Examples

```
console.log(`Removing obselete databases...`)
const databases = await indexedDB.databases()
for (const eachDatabase in databases) {
  if (eachDatabase.name !== DB_NAME) indexedDB.deleteDatabase(eachDatabase.name)
}
```

## Checklist

My suggestion meets these guidelines:

* [X] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [X] This wouldn't change the runtime behavior of existing JavaScript code
* [X] This could be implemented without emitting different JS based on the types of the expressions
* [ ] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [X] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by thomaschampagne** [comment]
Indeed it's missing. I had to do this as temporary workaround

```typescript
declare global {
    interface IDBFactory {
        databases: () => Promise<{ name: string, version: number }[]>;
    }
}
```

**Comment by nathanblair** [comment]
It appears there was an issue opened related to this already, #37317. If you feel it is appropriate, @RyanCavanaugh , feel free to close it and defer to that issue.

It looks like a PR was made by @fritzschoff, but they made changes to some generated files and the PR ended up closed without merge.

I've got the changes done myself to the lib/dom file (not the generated one), but it seems this:

a) would need coordinated with the TS-Generated repo? To get the generated declaration files updated from the lib/dom file changes made in my local
b) is being held up potentially for reasons related to how early-adoption it is. I can confirm it works in FF v75, at least. And definitely Chrome/Chromium.

I'll throw up my PR nonetheless and if no one likes it yet, feel free to cut it down :D

**Comment by fritzschoff** [comment]
I guess Firefox is the bottleneck here

**Comment by nathanblair** [comment]
It looks like my PR got auto-closed for a similar reason as yours, @fritzschoff. Even though I didn't even touch the /src/lib/*.generated* files?

So, :shrug: I'm over it.

**Comment by t3chguy** [comment]
This got added

![image](https://user-images.githubusercontent.com/2403652/234679026-e75c2f45-9a06-409d-acf2-e83c0c8f4a92.png)

**Comment by jakebailey** [comment]
As noted above, this was fixed in #44684.
